physics/continuous_trajectory_body.hpp

```diff
@@ -0,0 +1,228 @@
+﻿#pragma once
+
+#include <algorithm>
+
+#include "physics/continuous_trajectory.hpp"
+
+namespace principia {
+namespace physics {
+
+namespace {
+
+int const kMaxDegree = 17;
+int const kMinDegree = 3;
+
+// Only supports 8 divisions for now.
+int const kDivisions = 8;
+
+}  // namespace
+
+template<typename Frame>
+ContinuousTrajectory<Frame>::ContinuousTrajectory(Time const& step,
+                                                  Length const& low_tolerance,
+                                                  Length const& high_tolerance)
+    : step_(step),
+      low_tolerance_(low_tolerance),
+      high_tolerance_(high_tolerance),
+      degree_((kMinDegree + kMaxDegree) / 2) {
+  CHECK_LT(low_tolerance_, high_tolerance_);
+}
+
+template<typename Frame>
+bool ContinuousTrajectory<Frame>::empty() const {
+  return series_.empty();
+}
+
+template<typename Frame>
+Instant ContinuousTrajectory<Frame>::t_min() const {
+  CHECK(!empty()) << "Empty trajectory";
+  return *first_time_;
```

Comment:
`first_time_` is set by `ForgetBefore` and may not be at the beginning of a series (the semantics of a *continuous* trajectory implies that you should be able to cut it at any place regardless of its implementation details).